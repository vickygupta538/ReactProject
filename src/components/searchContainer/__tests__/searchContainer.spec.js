import React from "react";
import renderer from "react-test-renderer";
import { testStore } from "../../../../test/utils";
import SearchContainer from "../index";
import { Provider } from "react-redux";

const setUp = (props = {}) => {
  const initialState = {};
  const store = testStore(initialState);
  const component = renderer.create(
    <Provider store={store}>
      <SearchContainer />
    </Provider>
  );
  return component;
};

describe("SearchContainer Component Test Cases", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it("should match snapshot renderer", () => {
    const wrapper = component.toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render without errors", () => {
    const wrapper = component.root.find(
      el => el.type === "section" && el.props["data-test"] === "SearchContainerComponent"
    );

    expect(wrapper).toBeTruthy();
  });
});
