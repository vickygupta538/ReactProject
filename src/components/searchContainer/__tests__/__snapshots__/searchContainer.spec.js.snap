// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer Component Test Cases should match snapshot renderer 1`] = `
<section
  className="searchContainer"
  data-test="SearchContainerComponent"
>
  <center>
    <div>
      Search for your favourite Star Wars characters in the search box below. Click on their names to find out more information about them.
    </div>
    <section
      className="searchView"
      data-test="SearchViewComponent"
    >
      <div>
        <strong>
          Search:
        </strong>
      </div>
      <input
        className="searchTextBox"
        onChange={[Function]}
        type="input"
      />
    </section>
  </center>
</section>
`;
